frontend/src/components/configmap/List.tsx

```diff
@@ -1,13 +1,32 @@
 import { useTranslation } from 'react-i18next';
-import ConfigMap from '../../lib/k8s/configMap';
+import ConfigMap, { KubeConfigMap } from '../../lib/k8s/configMap';
+import { CreateResourceButton } from '../common/CreateResourceButton';
 import ResourceListView from '../common/Resource/ResourceListView';
 
+const BASE_EMPTY_CONFIG_MAP: KubeConfigMap = {
+  apiVersion: 'v1',
+  kind: 'ConfigMap',
+  metadata: {
+    creationTimestamp: '2023-04-27T20:31:27Z',
+    name: 'my-pvc',
+    namespace: 'default',
+    resourceVersion: '1234',
+    uid: 'abc-1234',
+  },
+  data: {},
+};
+
 export default function ConfigMapList() {
   const { t } = useTranslation(['glossary', 'translation']);
 
   return (
     <ResourceListView
       title={t('glossary|Config Maps')}
+      headerProps={{
+        titleSideActions: [
+          <CreateResourceButton resource={BASE_EMPTY_CONFIG_MAP} resourceName="ConfigMap" />,
```

Comment:
You can add this button as default, and the ResourceListView is already passed the resourceClass, so if we have the base object declared the class as I suggested, then it should be easy to check whether the class has a base object and use a generic create resource button for it if it does.